graph2tac/loader/predict_server.py

```diff
@@ -155,18 +157,23 @@ def measure_update_def_time(self, n_def_clusters_updated):
         self.n_def_clusters_updated += n_def_clusters_updated
         self.update_def_time += t1 - t0
 
-    def measure_t_coq_start(self):
-        if self.t0_coq >= 0:
+    @contextmanager
+    def measure_t_coq_send(self):
+        t0 = time.time()
+        yield
+        t1 = time.time()
+        self.t_coq_send += t1-t0
+        if self.t0_coq_receive >= 0:
             logger.warning("(coq_start) Nesting theorems in LoggingCounters, likely causing to misleading summaries")
-        self.t0_coq = time.time()
+        self.t0_coq_receive = t1
     def measure_t_coq_finish(self):
-        t1_coq = time.time()
-        if self.t0_coq < 0:
+        t1_coq_receive = time.time()
+        if self.t0_coq_receive < 0:
             logger.warning("(coq_finish) Nesting theorems in LoggingCounters, likely causing misleading summaries")
             return
         self.n_coq += 1
```

Comment:
Isn't there an off-by-one error with using the same counter for receive and coq_send.  We will send one more message than we receive, right (since we aren't counting the time to receive the original message)?

Reply:
I actually also count receiving the "iteration termination" message